dissect/target/plugins/os/windows/network.py

```diff
@@ -0,0 +1,317 @@
+                    # if no NetCfgInstanceId is found, skip this network interface
+                    continue
+
+                # Extract the network device configuration for given interface id
+                if (config := self._extract_network_device_config(net_cfg_instance_id)) is None:
+                    # if no configuration is found, skip this network interface
+                    continue
+
+                # Update the device info with the extracted configuration
+                device_info.update(config)
+
+                # Extract the network device name for given interface id
+                device_info["name"] = self._extract_network_device_name(net_cfg_instance_id)
+
+                # Extract the rest of the device information
+                device_info["mac"] = _try_value(subkey, "NetworkAddress")
+                device_info["vlan"] = _try_value(subkey, "VlanID")
+
+                if timestamp := _try_value(subkey, "NetworkInterfaceInstallTimestamp"):
+                    device_info["first_connected"] = wintimestamp(timestamp)
+
+                if type_device := _try_value(subkey, "*IfType"):
+                    device_info["type"] = IfTypes(int(type_device)).name
+
+                # Yield the device information
+                yield WindowsInterfaceRecord(
+                    **device_info,
+                    source="registry",
+                    last_connected=None,
+                    enabled=None,
+                    _target=self.target,
+                )
+
+    def _extract_network_device_config(self, interface_id: str) -> dict[str, str | list] | None:
+        config = {"search_domain": None, "gateway": [], "ip": [], "dns": [], "network": [], "metric": None}
+
+        try:
+            keys = self.target.registry.key(self.REGISTRY_KEY_INTERFACE + interface_id)
+        except RegistryKeyNotFoundError:
+            return None
+
+        if not len(keys):
+            return None
+
+        # Static lookups and assignments
+        for name in ["DhcpDefaultGateway", "DefaultGateway"]:
+            if (value := _try_value(keys, name)) not in ["", "0.0.0.0", None, []]:
+                config["gateway"].extend(value)
+
+        for name in ["DhcpIPAddress", "IPAddress"]:
+            if (value := _try_value(keys, name)) not in ["", "0.0.0.0", None]:
+                config["ip"].append(value)
+
+        for name in ["DhcpNameServer", "NameServer"]:
+            if (value := _try_value(keys, name)) not in ["", "0.0.0.0", None]:
+                config["dns"].append(value)
+
+        for name in ["DhcpSubnetMask", "SubnetMask"]:
+            if (value := _try_value(keys, name)) not in ["", "0.0.0.0", None]:
+                config["network"].append(value)
+
+        for name in ["DhcpDomain", "Domain"]:
+            if (value := _try_value(keys, name)) not in ["", None]:
+                config["search_domain"] = value
```

Comment:
`Domain` can override `DhcpDomain`, is that intentional (in comparison with the other ones)?